spec/unit/face/generate_spec.rb

```diff
@@ -220,6 +220,20 @@ module Puppet
         expect(stat_before <=> stats_after).to be(0)
       end
 
+      it 'overwrites all files if ruby files in lib/puppet_x/ are updated' do
+        # create them (first run)
+        puppet_x_lib = File.join(m1, 'lib', 'puppet_x', 'foo', 'library.rb')
+        Puppet::FileSystem.mkpath(puppet_x_lib)
+        genface.types
+        stats_before = [Puppet::FileSystem.stat(File.join(outputdir, 'test1.pp')), Puppet::FileSystem.stat(File.join(outputdir, 'test2.pp'))]
+        # Sorry. The sleep is needed because Puppet::FileSystem.touch(<path>, :mtime => Time.now + 1000) doesn't work on Windows.
+        sleep(1)
```

Comment:
That is surprising because our filesystem code calls `FileUtils.touch` which eventually passes the `mtime` through to https://github.com/ruby/ruby/blob/536649f819ed8f2bb0f8f44b1a0ca5c6d1753b24/win32/win32.c#L7692 where `SetFileTime` is documented in https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-setfiletime

Reply:
I tried a build just now that uses:
```ruby
Puppet::FileSystem.touch(puppet_x_lib, :mtime => Time.now + 1000)
```

It failed in the jruby spec tests: https://github.com/puppetlabs/puppet/actions/runs/6882098596/job/18719877970

From what I remember when I first made this PR, the `sleep(1)` was the only way to get the tests to pass.
Though now that I look at it, the comment about "doesn't work on Windows" might be wrong... it's actually failing on Jruby and the Windows tests are being skipped because of that.